services/brig/src/Brig/API/Public.hs

```diff
@@ -1063,12 +1065,47 @@ activate (Public.Activate tgt code dryrun)
     respond (Just ident) first = ActivationResp $ Public.ActivationResponse ident first
     respond Nothing _ = ActivationRespSuccessNoIdent
 
--- Verification
-
-sendVerificationCode :: (Handler r) ()
-sendVerificationCode =
-  case Public.TeamFeatureSndFPasswordChallengeNotImplemented of
-    _ -> pure ()
+sendVerificationCode :: Public.SendVerificationCode -> (Handler r) ()
+sendVerificationCode req = do
+  let action = Public.svcAction req
+  -- TODO(leif): guard as long as not all of the actions are implemented
+  -- remove when all done
+  case action of
+    Public.GenerateScimToken -> pure ()
+    Public.Login -> do
+      let email = Public.svcEmail req
+      mbUserId <- lift $ UserKey.lookupKey $ UserKey.userEmailKey email
+      mbAccount <- lift $ join <$> Data.lookupAccount `traverse` mbUserId
+      featureStatus <- getFeatureStatus mbAccount
+      case (mbUserId, featureStatus) of
+        (Just userId, Public.TeamFeatureEnabled) -> do
+          gen <- Code.mk6DigitGen $ Code.ForEmail email
+          mbPendingCode <- lift $ Code.lookup (Code.genKey gen) (scope action)
+          case mbPendingCode of
+            Nothing -> do
+              Timeout timeout <- setActivationTimeout <$> view settings
+              code <-
+                Code.generate
```

Comment:
This looks like duplication; similar code exists for password reset, account deletion, and provider creation (grep e.g. for `genKey`). Can that be refactored so we don't introduce very similar logic one more time?